src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryLayouts.java

```diff
@@ -61,6 +62,239 @@ private MemoryLayouts() {
 
+    private static final VarHandle byte_LE_handle = indexedHandle(BITS_8_LE, byte.class);
+    private static final VarHandle char_LE_handle = indexedHandle(BITS_16_LE, char.class);
+    private static final VarHandle short_LE_handle = indexedHandle(BITS_16_LE, short.class);
+    private static final VarHandle int_LE_handle = indexedHandle(BITS_32_LE, int.class);
+    private static final VarHandle float_LE_handle = indexedHandle(BITS_32_LE, float.class);
+    private static final VarHandle long_LE_handle = indexedHandle(BITS_64_LE, long.class);
+    private static final VarHandle double_LE_handle = indexedHandle(BITS_64_LE, double.class);
+
+    /**
+     * Read a byte from given address and offset, with byte order set to {@link ByteOrder#LITTLE_ENDIAN}.
+     * <p>
+     * This is equivalent to the following code:
+     * <blockquote><pre>{@code
+    VarHandle handle = MemoryHandles.withStride(BITS_8_LE.withBitAlignment(8).varHandle(byte.class), 1L);
+    byte value = (byte)handle.get(addr, offset);
+     * }</pre></blockquote>
+     *
+     * @param addr base address.
+     * @param offset offset (relative to {@code addr}). The final address of this read operation can be expressed as {@code addr.addOffset(offset)}.
+     * @return a byte value read from {@code addr} at the offset specified by {@code index}.
+     */
+    public static byte getByte_LE(MemoryAddress addr, long offset) {
+        return (byte)byte_LE_handle.get(addr, offset);
+    }
+
+    /**
+     * Writes a byte at given address and offset, with byte order set to {@link ByteOrder#LITTLE_ENDIAN}.
+     * <p>
+     * This is equivalent to the following code:
+     * <blockquote><pre>{@code
+    VarHandle handle = MemoryHandles.withStride(BITS_8_LE.withBitAlignment(8).varHandle(byte.class), 1L);
+    handle.set(addr, offset, value);
+     * }</pre></blockquote>
+     * @param addr base address.
+     * @param offset offset (relative to {@code addr}). The final address of this read operation can be expressed as {@code addr.addOffset(offset)}.
+     * @param value the byte value to be written.
+     */
+    public static void setByte_LE(MemoryAddress addr, long offset, byte value) {
+        byte_LE_handle.set(addr, offset, value);
+    }
+
+    /**
+     * Read a char from given address and offset, with byte order set to {@link ByteOrder#LITTLE_ENDIAN}.
+     * <p>
+     * This is equivalent to the following code:
+     * <blockquote><pre>{@code
+    VarHandle handle = MemoryHandles.withStride(BITS_16_LE.withBitAlignment(8).varHandle(char.class), 1L);
+    char value = (char)handle.get(addr, offset);
+     * }</pre></blockquote>
+     * @param addr base address.
+     * @param offset offset (relative to {@code addr}). The final address of this read operation can be expressed as {@code addr.addOffset(offset)}.
+     * @return a char value read from {@code addr} at the offset specified by {@code index}.
```

Comment:
Here, and similarly elsewhere:
```suggestion
     * @return a char value read from {@code addr} at the offset specified by {@code offset}.
```